src/components/PostDetails.tsx

```diff
@@ -1,106 +1,125 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+import { Errors } from '../types/Errors';
+import { getComments, deleteComment } from '../api/comments';
+
+type Props = {
+  post: Post;
+  newCommentCreating: boolean;
+  setNewCommentCreating: (value: boolean) => void;
+};
+
+export const PostDetails: React.FC<Props> = ({
+  post,
+  newCommentCreating,
+  setNewCommentCreating,
+}) => {
+  const { id, title, body } = post;
+
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isCommentsLoading, setIsCommentsLoading] = useState(false);
+  const [currentError, setCurrentError] = useState<Errors | null>(null);
+
+  const handleDeleteComment = (comment: Comment) => {
+    setComments(comments.filter(c => c.id !== comment.id));
+
+    deleteComment(comment.id).catch(() => {
+      setComments([...comments, comment]);
+      setCurrentError(Errors.CommentDeleting);
```

Comment:
When restoring the comment in case of a delete failure, consider providing user feedback to indicate that the deletion failed and the comment was restored.